test/e2e/specs/default-config/recordedit/immutable-inputs.spec.ts

```diff
@@ -0,0 +1,210 @@
+        value: 'Default for foreign_key_disabled column', disabled: true
+      },
+      { name: 'int_disabled', displayname: 'int_disabled', type: RecordeditInputType.INT_4, value: '20', disabled: true },
+      { name: 'float_disabled', displayname: 'int_disabled', type: RecordeditInputType.NUMBER, value: '93.2182', disabled: true },
+      { name: 'boolean_disabled', displayname: 'boolean_disabled', type: RecordeditInputType.BOOLEAN, value: 'false', disabled: true },
+      { name: 'date_disabled', displayname: 'date_disabled', type: RecordeditInputType.DATE, value: '2014-05-12', disabled: true },
+      {
+        name: 'timestamp_disabled', displayname: 'timestamp_disabled', type: RecordeditInputType.TIMESTAMP, disabled: true,
+        value: { date_value: '2012-06-22', time_value: '18:36:00' }
+      },
+      {
+        name: 'timestamptz_disabled', displayname: 'timestamptz_disabled', type: RecordeditInputType.TIMESTAMP, disabled: true,
+        value: { date_value: '2010-06-13', time_value: '17:22:00' }
+      },
+      {
+        name: 'json_disabled', displayname: 'json_disabled', type: RecordeditInputType.JSON, disabled: true,
+        value: JSON.stringify(98.786)
+      },
+      { name: 'color_rgb_hex_disabled', displayname: 'text_disabled', type: RecordeditInputType.COLOR, value: '#654321', disabled: true },
+    ]
+  }
+
+}
+
+test.describe('Immutable and Generated columns', () => {
+  test.describe.configure({ mode: 'parallel' });
+
+  test('Create mode', async ({ page, baseURL }, testInfo) => {
+    await test.step('open recordedit page', async () => {
+      await page.goto(`${baseURL}/recordedit/#${getCatalogID(testInfo.project.name)}/${testParams.schemaTable}`);
+      await RecordeditLocators.waitForRecordeditPageReady(page);
+    });
+
+    for await (const col of testParams.create.columns) {
+      await test.step(`${col.displayname}`, async () => {
+        await test.step(`display ${col.disabled ? 'disabled' : ''} input with proper value`, async () => {
+          await testInputValue(page, 1, col.name, col.displayname, col.type, col.disabled, col.value);
+        });
+
+        if (col.autoGenerated) {
+          await test.step('display the "Automatically generated" placeholder', async () => {
+            switch (col.type) {
+              case RecordeditInputType.TIMESTAMP:
+                const props = RecordeditLocators.getTimestampInputsForAColumn(page, col.name, 1);
+                await testAutomaticallyGenerated(props.time, col.displayname);
+                await testAutomaticallyGenerated(props.date, col.displayname);
+                break;
+              case RecordeditInputType.FILE:
+              case RecordeditInputType.FK_DROPDOWN:
+              case RecordeditInputType.FK_POPUP:
+              case RecordeditInputType.BOOLEAN:
+                await expect.soft(RecordeditLocators.getInputPlaceholderMessage(page, col.name, 1)).toHaveText('Automatically generated');
+                // TODO
```

Comment:
Was this TODO meant for something else?

Reply:
I added this before adding the `getInputPlaceholderMessage` locator and forgot to remove it once it was implemented.